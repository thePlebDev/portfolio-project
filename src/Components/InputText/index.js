import React,{useState} from 'react';
import styled from 'styled-components'

const Input = styled.input`
  width:70%;
  color:#0f141e;
  padding-top:20px;
  border:none;
  transition:all .2s;
  font-size:20px;
  border-bottom:${props =>props.error ? '1px solid red':'1px solid #0f141e'};
  outline:none;



  @media only screen and (min-width: 1200px){
    width:53%;
  }
`

const Label = styled.label`
  transition:all .3s;
  left:0;
  bottom:${props=>props.state? '50%':'0'};
  position:absolute;
  font-size:${props=>props.state? '12px':'16px'};
  text-transform:uppercase;
  color:${props =>props.error ? 'red':'#0f141e'};
  opacity:${props=>props.state? '1':'0.5'};

`


const InputText =(

  {name='enter name prop',value='enter value prop',
  onChange={function(){console.log('enter onChange prop')}},error='enter error prop',type='text'}
              )=>{

  const [state,setState] = useState(false)

  const handleClick =(e)=>{
    if(e.target.value){
      setState(true)
    }
    else{
      setState(!state)
    }

  }

  return(
    <div style={{position:'relative'}}>
      <Input type={type} error={error} id={name} autocomplete="off" onFocus={(e)=>handleClick(e)} data-testid="input"
        onBlur={(e)=>handleClick(e)} value={value} onChange={(e)=>onChange(e)} onClick={(e)=>handleClick(e)} name={name}>
      </Input>
      <Label state={state} error={error} data-testid="label" className="label" for={name}>{name}</Label>
    </div>
  )
}

export default InputText
