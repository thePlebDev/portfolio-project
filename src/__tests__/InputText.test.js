import React from 'react';

import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import InputText from '../Components/InputText'


describe('testing the InputText component',()=>{
  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<InputText/>)
  })

  it('should render a proper snapshot',()=>{
    const tree = renderer.create(<InputText />).toJSON()
    expect(tree).toMatchSnapshot()

  })
  it('should render everything properly',()=>{
    expect(wrapper.find('[data-testid="input"]').length).toBe(1)
    expect(wrapper.find('[data-testid="input"]').length).toBe(1)
  })
  it('should test the default props',()=>{
    const defaultError ='enter error prop';
    const defualtName ='enter name prop';
    const defaultValue = 'enter value prop'
    const input = wrapper.find('[data-testid="input"]')
    const label = wrapper.find('[data-testid="label"]')

    expect(input.props().name).toBe(defualtName)
    expect(input.props().value).toBe(defaultValue)
    expect(input.props().error).toBe(defaultError)

    expect(label.props().state).toBe(false)
    expect(label.props().error).toBe(defaultError)
    expect(label.props().for).toBe(defualtName)
  })

  it('should test the actions',()=>{
    const me = shallow(<InputText/>)
    const input = me.find('[data-testid="input"]')
    const label = me.find('[data-testid="label"]')
    input.simulate('click',{target:{value:'another'}})
    console.log(label.props().state)
  })

})
