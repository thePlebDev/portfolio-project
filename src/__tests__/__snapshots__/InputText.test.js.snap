// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the InputText component should render a proper snapshot 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <input
    className="sc-bdnylx dTAPuY"
    data-testid="input"
    id="enter name prop"
    name="enter name prop"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="text"
    value="enter value prop"
  />
  <label
    className="sc-gtssRu koWzil label"
    data-testid="label"
    for="enter name prop"
  >
    enter name prop
  </label>
</div>
`;
